### DISCUSSION TITLE
Replace `r2d2` (non-async) with `sqlx` (async)

### DISCUSSION THREAD

**Discussion by mickvandijke** [torrust/torrust-tracker#616] [Proposals]

**Comment by josecelano** [comment]
Hi @WarmBeer @da2ce7, we should consider more alternatives like:

- https://diesel.rs/ (`mysql`, `postgresql`, `sqlite`)
- https://github.com/prisma/quaint (`mysql`, `postgresql`, `sqlite`, `mssql`)

I think we should decide first which features we want to have:

- ORM
- DB migrations
- Support for other databases, not only `mysql` and `sqlite`
- Performance
- Async
- ...

I do not think having support for multiple DBs or an ORM is necessary for this project.

I'm assuming:

- The DB schema is not going to change often, but DB migrations are still convenient.
- The persistence is minimal.

From my point of view, I would:

- Switch to `sqlx` if it's faster.
- Include DB migrations.
- Split the `Database` trait into more independent repositories, like the contexts we have for the API: `stats`, `torrents`, `whitelist`, `auth_keys`.
